packages/openapi-typescript/src/utils.ts

```diff
@@ -106,15 +106,31 @@ export function comment(text: string, indentLv?: number): string {
 /** handle any valid $ref */
 export function parseRef(ref: string): { filename: string; path: string[] } {
   if (typeof ref !== "string") return { filename: ".", path: [] };
-  if (!ref.includes("#")) return { filename: ref, path: [] };
-  const [filename, path] = ref.split("#");
-  return {
-    filename: filename || ".",
-    path: path
-      .split("/") // split by special character
-      .filter((p) => !!p && p !== "properties") // remove empty parts and "properties" (gets flattened by TS)
-      .map(decodeRef), // decode encoded chars
-  };
+
+  // OpenAPI $ref
+  if (ref.includes("#/")) {
```

Comment:
Was able to get a 2x speedup of `parseRef` out of refactoring it a little. Which is something, because this is used quite a lot.